File: Plugins/Aspose_Slides_Java_for_Ruby/lib/asposeslidesjava/Presentation/convertingtohtml.rb
module Asposeslidesjava
  module ConvertingToHtml
    def initialize()
        # Converting Presentation to HTML format
        convert_to_html()
    end

    def convert_to_html()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate a Presentation object that represents a PPTX file
        pres = Rjb::import('com.aspose.slides.Presentation').new(data_dir + "Aspose.pptx")

        html_opt = Rjb::import('com.aspose.slides.HtmlOptions').new
        html_formatter = Rjb::import('com.aspose.slides.HtmlFormatter')
        html_opt.setHtmlFormatter(html_formatter.createDocumentFormatter("",false))

        # Saving the presentation to HTML format
        save_format = Rjb::import('com.aspose.slides.SaveFormat')
        pres.save(data_dir + "Aspose.html", save_format.Html, html_opt)

        puts "Document has been converted, please check the output file."
    end    
  end
end

File: Plugins/Aspose_Slides_Java_for_PHP/src/aspose/slides/WorkingWithPresentation/ConvertingToHtml.php
<?php

namespace Aspose\Slides\WorkingWithPresentation;

use com\aspose\slides\Presentation as Presentation;
use com\aspose\slides\SaveFormat as SaveFormat;
use com\aspose\slides\HtmlOptions as HtmlOptions;
use com\aspose\slides\HtmlFormatter as HtmlFormatter;


class ConvertingToHtml {

    public static function run($dataDir=null)
    {

        # Converting Presentation to HTML format
        ConvertingToHtml::convert_to_html($dataDir);
    }

    public static function convert_to_html($dataDir=null)
    {

        # Instantiate a Presentation object that represents a PPTX file
        $pres = new Presentation($dataDir . "Aspose.pptx");
        $html_opt = new HtmlOptions();
        $html_formatter = new HtmlFormatter();
        $html_opt->setHtmlFormatter($html_formatter->createDocumentFormatter("",false));

        # Saving the presentation to HTML format
        $save_format = new SaveFormat();
        $pres->save($dataDir . "Aspose.html", $save_format->Html, $html_opt);

        print "Document has been converted, please check the output file.";

    }

}